contrib/update-my-copyright.pl

```diff
@@ -67,6 +67,9 @@
 my $my_search_name = "Cisco";
 my $my_formal_name = "Cisco Systems, Inc.  All rights reserved.";
 
+# Protected directories
+my @protected = qw(pmi2x/pmix/ hwloc1113/hwloc/ libevent2022/libevent/);
```

Comment:
Should we make these regexps to make them a bit more future-proof?  Otherwise, we'll need to remember to edit this script whenever we update these components.

Eg:

```diff
diff --git a/contrib/update-my-copyright.pl b/contrib/update-my-copyright.pl
index bed0f0a..5b068e0 100755
--- a/contrib/update-my-copyright.pl
+++ b/contrib/update-my-copyright.pl
@@ -68,7 +68,12 @@ my $my_search_name = "Cisco";
 my $my_formal_name = "Cisco Systems, Inc.  All rights reserved.";
 
 # Protected directories
-my @protected = qw(pmi2x/pmix/ hwloc1113/hwloc/ libevent2022/libevent/);
+my @protected = qw(
+    opal\\/mca\\/pmi\\/pmix.+?\\/pmix\\/
+    opal\\/mca\\/hwloc\\/hwloc.+?\\/hwloc\\/
+    opal\\/mca\\/libevent\\/libevent.+?\\/libevent\\/
+    contrib\\/update-my-copyright.pl
+);
 
 # Override the defaults if some values are set in the environment
 $my_search_name = $ENV{OMPI_COPYRIGHT_SEARCH_NAME}
@@ -155,7 +160,7 @@ foreach my $f (@files) {
     # the copyright
     my $ignore = 0;
     foreach my $p (@protected) {
-        if (index($f, $p) != -1) {
+        if (eval("\$f =~ /$p/")) {
             quiet_print "Ignoring protected file $f\n";
             $ignore = 1;
             last;
```